cli/cli/syntho_cli.py

```diff
@@ -726,6 +741,102 @@ def dc_deployment(
         sys.exit(1)
 
 
+@dc.command(name="update", help="Updates the deployment with a given version")
+@click.option(
+    "--deployment-id",
+    type=str,
+    help="Specify the deployment id to be status checked",
+    required=False,
+    default="",
+    callback=validate_dc_deployment_id,
+)
+@click.option("--new-version", type=str, help=("Specify a version for Syntho stack."), required=True)
+def dc_deployment_update(deployment_id: str, new_version: str):
+    try:
+        validate_version(new_version)
+    except click.BadParameter as exc:
+        raise click.UsageError(str(exc)) from exc
+
+    deployment = dc_deployment_manager.get_deployment(scripts_dir, deployment_id)
+    if not deployment:
+        raise click.UsageError(f"There is no release found with the given deployment id ({deployment_id})")
+
+    current_version = deployment["version"]
+    releases = get_releases(with_compatibility=current_version)
+    versions = [release.get("name") for release in releases if release["name"] != current_version]
+    versions_str = "\n".join(versions)
+
+    new_version_major_ver, _, _ = new_version.split(".")
+    if not current_version.startswith(new_version_major_ver):
+        raise click.UsageError(
+            f"Given version ({new_version}) is not compatible "
+            f"within the current stack ({current_version}).\nCompatible releases:\n{versions_str}"
+        )
+
+    if current_version == new_version:
+        raise click.UsageError(
+            f"Given version ({new_version}) is already deployed "
+            f"within the current stack ({current_version}).\nCompatible releases:\n{versions_str}"
+        )
```

Comment:
This section of the `dc_deployment_update` function is exactly the same as it is in the kubernetes version. Could we separate in a different function to be used in both sides?

Reply:
In case we make any changes to that logic that needs to propagate to both Kubernetes and Docker Compose

Reply:
Added tech debt item for this